src/libraries/System.Net.Sockets/tests/FunctionalTests/TcpListenerTest.cs

```diff
@@ -252,6 +252,36 @@ public void ExclusiveAddressUse_SetStartAndStopListenerThenRead_ReadSuccessfully
             Assert.True(listener.ExclusiveAddressUse);
         }
 
+        [Fact]
+        public void EndAcceptSocket_WhenStopped_ThrowsObjectDisposedException()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.Start();
+
+            IAsyncResult iar = listener.BeginAcceptSocket(callback: null, state: null);
+
+            // Give some time for the underlying OS operation to start:
+            Thread.Sleep(50);
```

Comment:
We often see tests fail because of slow test machines, I wonder whether this will reliably be long enough.

Reply:
In this test the next call should throw `ObjectDisposedException` independently from timing, so slow machines should not bring an issue hopefully.